### ISSUE TITLE
feat(tracking): Madara core

### ISSUE THREAD

**Issue by Trantorian1** [KasarLabs/deoxys#413]
# About

This issue serves as the main entry point towards new contributions into the stable branch of Madara: it is designed to track the progress and priority of implementing core features into the codebase.

## `feat`: RPC v0.8.0 :red_circle:

> [!NOTE]
> Latest version of the Startknet RPC specs. Boilerplate is already in place, only the code for method implementations needs to be written.

- [ ]  #335 _(underway)_
- [x] #443

### Priority

This is relatively urgent and the main goal for stable Madara at the moment. RPC v0.8.0 is slated for testnet integration at the end of December: we would like node operators to be able to start experimenting with these methods by then.

## `feat`: Block Production Restart Refactor :yellow_circle:
> [!NOTE]
> In case of restart, in current approach madara re-executes the transactions which shouldn't be the case. Detailed refactor proposal can be found in the issue listed.

- [x] #458 

## `feat`: reorgs :yellow_circle:

> [!NOTE]
> Add support for l1 and l2 reorgs. Expect a significant amount of rewrite of the l2 and l1 sync logic. I suggest you see about this with @cchudant as they are working on a new shared block import pipeline for l2 and local block production which might affect this.

- [ ] #202 (_underway_)

### Priority

This is a much needed feature and needs urgent work as well, but due to its scope should probably be addressed on a longer time frame than RPC v0.8.0

## `perf`: RPC :yellow_circle:

> [!NOTE]
> Current implementation of RPC methods suffers from poor optimization. There is a lot of room for improvement though so the ceiling for is low.

- [ ] #378 

### Priority

We need to consider that most of the time a node spends is answering RPC request, so this is very important but should still probably come after #335.

## `enhancements`: new features :green_circle:

> [!NOTE]
> New features which would make the handling of Madara more easy for end users.

- [x] #285 
- [ ] #223

### Priority

Take up these tasks if all other issues have been resolved.

## `refactor`: code Cleanup :green_circle:

> [!NOTE]
> Low priority refactoring of the codebase, no functional changes. These are features that are nice to have and help will help in the maintainability of Madara on the long term.

- [ ] #412 
- [x] #387
- [x] #293 
- [ ] #231 _(underway)_
- [ ] #422

### Priority

Take up these tasks if your have some spare time.

**Comment by heemankv** [comment]
Status update from codebase review on 2026-05-05.

Classification: Still relevant.

Current state:
Still a live umbrella because multiple child/core tickets remain open.

Keeping this issue open based on the current tree.
